### ISSUE TITLE
Automatized creation of functional region of interests (ROIs) using Python tools

### ISSUE THREAD

**Issue by ilkayisik** [ohbm/hackathon2019#41]
# Automatized creation of functional region of interests (ROIs) using Python tools

Ilkay Isik [ORCID](https://orcid.org/0000-0002-1652-9297)
## Project Description
Functionally defining regions of interests is a common methodology in cognitive neuroscience due to the greater sensitivity and higher functional resolution it provides over group-based methods (Nieto-Castanon and Fedorenko, 2012). In this approach, a set of functional regions are defined in each individual using a localizer contrast targeting the cognitive process of interest (e.g. fusiform face area (FFA) obtained by contrasting Faces vs Objects).
However, there is not a commonly accepted and automatized way of delineating and selecting these ROIs. The traditional method is to select subject specific ROIs by examining the activation maps for the localizer contrast and decide which voxels to include manually using anatomy knowledge as a guide. However, even expert coders might disagree due to the high individual variability. Furthermore, when these ROIs happen to be located close to each other, it is not very straightforward how to draw the border between them.

Fedorenko et al (2010) and Julian et al (2012) were concerned about these problems
and they offered the following steps in order to automate the creation of ROIs algorithmically:

<img width="847" alt="Fedorenko_2010" src="https://user-images.githubusercontent.com/22478219/55884054-b07a9380-5ba7-11e9-9dcd-110fed4f6f37.png">

###### Image used with permission from Dr. Evelina Fedorenko

The authors have used Matlab to accomplish these goals.
In this project, we aim to use Python tools and create a Python package which automatically creates functional region of interests.
So, this project itself is not a new idea but I believe this will be a great learning experience for me and for anyone who wants to join and contribute.

## Skills required to participate
- Experience in programming in Python
- Familiarity with the functional localizer literature (not necessary)

## Integration
You can contribute to this project by helping with
- Programming
- Pick and apply an image segmentation algorithm
- Documenting [improve docstrings, better commenting]

## Preparation material
- Whitfield-Gabrieli, S., Fedorenko, E., Nieto-Castañón, A., Hsieh, P.-- J., & Kanwisher, N. (2010). New Method for fMRI Investigations of Language: Defining ROIs Functionally in Individual Subjects. Journal of Neurophysiology, 104(2), 1177–1194. https://doi.org/10.1152/jn.00032.2010

- Julian, J. B., Fedorenko, E., Webster, J., & Kanwisher, N. (2012). An algorithmic method for functionally defining regions of interest in the ventral visual pathway. NeuroImage, 60(4), 2357–2364. https://doi.org/10.1016/j.neuroimage.2012.02.055

- Nieto-Castañón, A., & Fedorenko, E. (2012). Subject-specific functional localizers increase sensitivity and functional resolution of multi-subject analyses. NeuroImage, 63(3), 1646–1669. https://doi.org/10.1016/j.neuroimage.2012.06.065

## Link to your GitHub repo
[ProjectGitHubRepo](https://github.com/ilkayisik/PyFuncROI)

## Communication
Join  [![slack_brainhack_3](https://user-images.githubusercontent.com/6297454/47951457-5b37b780-df61-11e8-9d77-7b5a4c7af875.png)](https://brainhack-slack-invite.herokuapp.com/) and find channel __#pyfuncroi__

**Comment by r03ert0** [comment]
That's a fantastic project @ilkayisik ! :+1:
- The link to the GitHub repo gives a 404 error
- You could maybe link to the introductory material in Python programming from our Tutorial list? https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#python

**Comment by r03ert0** [comment]
Maybe you could also link to the section on "Documenting projects and code": https://github.com/ohbm/hackathon2019/blob/master/Tutorial_Resources.md#documenting-projects-and-code

**Comment by ilkayisik** [comment]
Thanks so much @r03ert0.
I fixed the error and also added the links you suggested

**Comment by Conxz** [comment]
A very nice project. Here is a project I think with a similar aim. https://github.com/BNUCNL/FreeROI

**Comment by seldamat** [comment]
this sounds like a great project, would it be considered too far from the core idea by also including ROI definition using ICA on task-free (resting) data?

**Comment by ilkayisik** [comment]
not at all. I was originally thinking of ventral visual pathway ROIs but creation of ROIs from certain network nodes using a resting state analysis could be a nice addition.
